### ISSUE TITLE
Why is the accuracy rate stuck at 50%

### ISSUE THREAD

**Issue by be10nd** [NanFangXiJi/Steganalysis-LWENet#1]
First of all,why is the format of the valid path different from that of train?Why is there only one folder in the path to valid？
Second,Why is the accuracy rate only 50%, and there is no training at all!
```
lr is now 0.01
train Epoch: 0	avgLoss: 0.739817
valid set: Average loss: 0.7898, Accuracy: 100.0/200 (50.000000%)
best model saved at models/example_BEST.pth
lr is now 0.01
train Epoch: 1	avgLoss: 0.714590
valid set: Average loss: 0.7320, Accuracy: 100.0/200 (50.000000%)
best model saved at models/example_BEST.pth
lr is now 0.01
train Epoch: 2	avgLoss: 0.711945
valid set: Average loss: 0.7004, Accuracy: 100.0/200 (50.000000%)
best model saved at models/example_BEST.pth
lr is now 0.01
train Epoch: 3	avgLoss: 0.704947
valid set: Average loss: 0.7366, Accuracy: 100.0/200 (50.000000%)
lr is now 0.01
train Epoch: 4	avgLoss: 0.699339
valid set: Average loss: 0.6958, Accuracy: 100.0/200 (50.000000%)
best model saved at models/example_BEST.pth
```

**Comment by be10nd** [comment]
> First of all,why is the format of the valid path different from that of train?Why is there only one folder in the path to valid？ Second,Why is the accuracy rate of running out of the pig's trotters only 50%, and there is no training at all!
> 
> ```
> lr is now 0.01
> train Epoch: 0	avgLoss: 0.739817
> valid set: Average loss: 0.7898, Accuracy: 100.0/200 (50.000000%)
> best model saved at models/example_BEST.pth
> lr is now 0.01
> train Epoch: 1	avgLoss: 0.714590
> valid set: Average loss: 0.7320, Accuracy: 100.0/200 (50.000000%)
> best model saved at models/example_BEST.pth
> lr is now 0.01
> train Epoch: 2	avgLoss: 0.711945
> valid set: Average loss: 0.7004, Accuracy: 100.0/200 (50.000000%)
> best model saved at models/example_BEST.pth
> lr is now 0.01
> train Epoch: 3	avgLoss: 0.704947
> valid set: Average loss: 0.7366, Accuracy: 100.0/200 (50.000000%)
> lr is now 0.01
> train Epoch: 4	avgLoss: 0.699339
> valid set: Average loss: 0.6958, Accuracy: 100.0/200 (50.000000%)
> best model saved at models/example_BEST.pth
> ```
> 
> Between someone's unpaid work and writing in such detail,this beautiful girl reluctantly meets the requirements of a pig to write an issue to ask questions,if it can't be solved,this beautiful girl won't star.

**Comment by NanFangXiJi** [comment]
# About dataset

You will need to split your dataset manually. 
The standard directory structure should be as follows:
```
data/
├──cover/
└── stego/
```

This format can be used for training, validation, and test datasets.

Regarding the unique dataloader for training, it's designed to meet the requirements of `DatasetPair`, which is necessary for proper data augmentation.

# About training failure

To better assist you with the training issues, I'll need more detailed information about:
- Your training parameters
- Your dataset structure
- Any modifications you've made to the original implementation

**Comment by be10nd** [comment]
运行前需要修改train.ipynb的哪些地方，valid_path是不是包含了cover和stego的文件夹的路径

**Comment by NanFangXiJi** [comment]
You need to configure the dataset path before starting. Optionally, you can adjust various parameters within reasonable ranges to suit your needs.  

This implementation is designed for quick deployment—you can get started with minimal modifications. Simply set up the paths, and you're ready to train.

**Comment by be10nd** [comment]
我做了以下修改：
```
train_cover_path = "ds/cover"
train_stego_path = "ds/stego"
valid_path = "ds"
test_path = "ds"

model_save_path = "models/"
```
在此我用从原始数据集摘出来的一部分数据集ds文件夹，cover和stego分别只有一百张图片，此时训练集和测试集都用ds，测试代码可行性。
ds的目录结构与author所说的data相同。
```
torch.cuda.empty_cache()
batch_size = {'train': 32, 'valid': 32, 'test': 32}
lr = 0.01
weight_decay = 0.001
momentum = 0.9
log_interval = 50  # 多少个batch打印一次
save_interval = 10  # 多少个epoch保存一次
epochs = 10

train_title = "example"
```
参数部分我稍作修改，如上。
然而，有两个问题，第一是楼顶说过的正确率问题，更像是完全没有训练，其次，训练速度很慢。如上所述，共二百张图片，epoch为10轮，训练了26min却只完成了四轮。
其他参数并未修改

**Comment by NanFangXiJi** [comment]
# About training failure

This issue commonly stems from dataset-related problems. Potential causes include:

1. **Algorithm-architecture mismatch**:  
   The steganography algorithm used in your dataset may be incompatible with the current deep learning network architecture. However, it is unlikely, as experiences have shown strong stability for a wide range of algorithms.

2. **Dataset integrity issues**:  
   There might be errors in your dataset construction, such as identical pairs in cover and stego images. Please do check.

3. **Scale-configuration imbalance**:  
   Notably, your relatively small dataset size combined with an excessively large batch setting could be another contributing factor.

# About slow training

This is a lightweight network architecture that typically trains significantly faster than conventional models. The slow training speed you're experiencing may be attributed to the following factors:

**Critical Considerations for Training Efficiency:**

1. **VRAM Capacity Management**  
   Ensure your batch size configuration does not exceed available GPU memory. When the batch size is set too large, causing memory overutilization, training performance will degrade substantially.

2. **Recommended Configuration**  
   Based on empirical testing:
   - T100 GPUs with 16GB VRAM can comfortably handle batch sizes up to 25
   - If your GPU has comparable memory capacity (≤16GB), avoid using larger batch sizes

3. **Optimization Suggestion**  
   For optimal performance:
   - Start with smaller batches
   - Gradually increase while monitoring memory usage

**Comment by be10nd** [comment]
现在将batch_size改成1了，看似能够正常训练了，但是速度依旧很慢，并没有广告上宣传的那么快，差评！

**Comment by NanFangXiJi** [comment]
Setting the batch size to 1 represents another extreme case that should generally be avoided. 

For optimal training performance, I recommend:

>    For optimal performance:
>    
>    * Start with smaller batches
>    * Gradually increase while monitoring memory usage
